### ISSUE TITLE
Add the option to rotate the images

### ISSUE THREAD

**Issue by tognee** [LibrePhotos/librephotos#856]
**Describe the enhancement you'd like**
An option to rotate an image from the image view.

This could be by adding a new field to the Photos database (saving the exif value in the DB and allowing editing it through the frontend), or by editing the images directly and updating the thumbnails and data for that image automatically.

**Describe why this will benefit the LibrePhotos**
This will make the app more user friendly. Currently to rotate an image you need to edit the original file and start a library scan. (Not sure if it works, as it didn't for me)

**Additional context**
I think the DB method could be the best solution as it's not destructive, but it's up to debate so we can choose the best solution. 

I'm familiar with python and web development so I'm willing to help develop this feature if help is needed.

**Comment by tognee** [comment]
The way the app is designed it's better to save the Orientation exif tag in the database and update the tag in the file once updated. There is already a _save_metadata function that I can reuse for this. I should add a `_regenerate_thumbnail` function that deletes the old thumbnails images and recreates them.

From my tests I need to run `_calculate_aspect_ratio` again to show the image with the correct ratio in the web ui.
Are there any other functions that I should run to apply the change correctly?

**Comment by derneuere** [comment]
I would prefer if you would just call the function ```_rotate_image```, because that is what it does. You would then need to delete the old thumbnail, create a new one, recalculate the aspect ratio. It would also make sense to let the AI features like face recognition, clip_embeddings and scene detection run again, as rotating the image could change the result.

Add a new field, "orientation" to the photo model, which would then get changed in that function. Add the orientation field to the save_metadata function as a possible field that could be saved to file.

**Comment by tognee** [comment]
Are all those methods run on the thumbnails? That way the rotate function could just rotate the thumbnails after generation.

Also I can't find the `_rotate_image` function you're talking about. Searching in the repo I found a pull request that modifies it but that file doesn't seem to exist anymore.

My current implementation needs to have the tag saved on the file itself (as the generate thumbnail function of the image library doesn't have an option to provide a rotation value), rotating the thumbnail after generation could solve the issue.

That way the orientation field would be separated from the exif value and just used to generate the thumbnails.

**Comment by tognee** [comment]
The backend is implemented and working, the frontend is gonna be a bit of a problem.

I have a PoC that works (as it turns the image and generates the new thumbnails), but you cannot see the changes "in real time" as you need to refresh the page for the Photo grid and the reference of the image in Lightbox to update. Sometimes the image doesn't get refreshed after a page refresh (the aspect ratio does tho) so you see the image half cropped or doubled and you need to do a hard refresh (by pressing shift on the reload button) to see the correct image.

I've never used React so that will also be a problem for me, if anyone wants to help with this feature you're more than welcome.

**Comment by derneuere** [comment]
It could be needed to add that functionality to the underlying Lightbox. I think opening up a WIP PR in the frontend would be the most straight forward way others can help you :)

**Comment by tognee** [comment]
I'm planning to do that, thanks. I got busy with work and friends stuff.

I'll mention this issue on the PR when I have it ready :)

**Comment by derneuere** [comment]
Non-destructive rotation is implemented — rotate CW/CCW buttons in the lightbox toolbar, with a backend rotate endpoint and thumbnail regeneration.
